autorest/adal/token.go

```diff
@@ -619,9 +623,24 @@ func (spt *ServicePrincipalToken) refreshInternal(resource string) error {
 	if _, ok := spt.secret.(*ServicePrincipalMSISecret); ok {
 		req.Method = http.MethodGet
 		req.Header.Set(metadataHeader, "true")
-	}
-
-	resp, err := spt.sender.Do(req)
+		retries = append(retries, http.StatusNotFound,
+			// all 5xx
+			http.StatusInternalServerError,
+			http.StatusNotImplemented,
+			http.StatusBadGateway,
+			http.StatusServiceUnavailable,
+			http.StatusGatewayTimeout,
+			http.StatusHTTPVersionNotSupported,
+			http.StatusVariantAlsoNegotiates,
+			http.StatusInsufficientStorage,
+			http.StatusLoopDetected,
+			http.StatusNotExtended,
+			http.StatusNetworkAuthenticationRequired)
+	}
+
+	resp, err := autorest.SendWithSender(spt.sender, req,
+		autorest.DoRetryForStatusCodes(autorest.DefaultRetryAttempts, time.Second*1, retries...),
```

Comment:
Is there no way we can use the retry attempts and delay that the caller configs on the client?